pkg/artifactory/repositories.go

```diff
@@ -330,442 +332,523 @@ var repoTypesLikeGeneric = []string{
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		MinItems:    1,
-		MaxItems:    2,
-		Set:         schema.HashString,
-		Optional:    true,
-		Description: `Project environment for assigning this repository to. Allow values: "DEV" or "PROD"`,
-	},
-	"package_type": {
-		Type:     schema.TypeString,
-		Required: false,
-		Computed: true,
-		ForceNew: true,
-	},
-	"description": {
-		Type:     schema.TypeString,
-		Optional: true,
-	},
-	"notes": {
-		Type:     schema.TypeString,
-		Optional: true,
-	},
-	"includes_pattern": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Computed:    true,
-		Description: "List of artifact patterns to include when evaluating artifact requests in the form of x/y/**/z/*. When used, only artifacts matching one of the include patterns are served. By default, all artifacts are included (**/*).",
-	},
-	"excludes_pattern": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Computed:    true,
-		Description: "List of artifact patterns to exclude when evaluating artifact requests, in the form of x/y/**/z/*. By default no artifacts are excluded.",
-	},
-	"repo_layout_ref": {
-		Type:        schema.TypeString,
-		Optional:    true,
-		Computed:    true,
-		Description: "Repository layout key for the local repository",
-	},
-	"blacked_out": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Default:     false,
-		Description: "When set, the repository does not participate in artifact resolution and new artifacts cannot be deployed.",
-	},
-	"xray_index": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Default:     false,
-		Description: "Enable Indexing In Xray. Repository will be indexed with the default retention period. You will be able to change it via Xray settings.",
-	},
-	"priority_resolution": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Default:     false,
-		Description: "Setting repositories with priority will cause metadata to be merged only from repositories set with this field",
-	},
-	"property_sets": {
-		Type:        schema.TypeSet,
-		Elem:        &schema.Schema{Type: schema.TypeString},
-		Set:         schema.HashString,
-		Optional:    true,
-		Description: "List of property set name",
-	},
-	"archive_browsing_enabled": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Description: "When set, you may view content such as HTML or Javadoc files directly from Artifactory.\nThis may not be safe and therefore requires strict content moderation to prevent malicious users from uploading content that may compromise security (e.g., cross-site scripting attacks).",
-	},
-	"download_direct": {
-		Type:        schema.TypeBool,
-		Optional:    true,
-		Description: "When set, download requests to this repository will redirect the client to download the artifact directly from the cloud storage provider. Available in Enterprise+ and Edge licenses only.",
-	},
+func getBaseLocalRepoSchema(packageType string) map[string]*schema.Schema {
```

Comment:
You don't need this function at all - look at your implementation:

```go
return map[string]*schema.Schema{}
```
Just declare the map as
```go
var baseLocalRepoSchema = ...
```
and then in your merge function, merge in just the bit acount the default layout

Reply:
improved as suggested